tests/integrationv2/test_session_resumption.py

```diff
@@ -102,3 +100,133 @@ def test_session_resumption_s2n_client(managed_process, cipher, curve, protocol,
         assert results.exception is None
         assert results.exit_code == 0
         assert results.stdout.count(bytes("6 server accepts that finished".encode('utf-8')))
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_tls13_session_resumption_s2n_server(managed_process, cipher, curve, protocol, provider, certificate):
+    port = str(next(available_ports))
+
+    ticket_filename = 'session_ticket_' + port
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        insecure=True,
+        reconnect=False,
+        extra_flags = ['-sess_out', ticket_filename],
+        data_to_send = data_bytes(4096),
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.use_session_ticket = True
+    server_options.extra_flags = None
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+
+    # Client inputs stored session ticket to resume a session
+    assert os.path.exists(ticket_filename)
+    client_options.extra_flags = ['-sess_in', ticket_filename]
+
+    port = str(next(available_ports))
+    client_options.port = port
+    server_options.port = port
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    openssl_version = get_expected_openssl_version(protocol)
+    s2n_version = get_expected_s2n_version(protocol, OpenSSL)
+
+    # The client should have received a session ticket
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert b'Post-Handshake New Session Ticket arrived:' in results.stdout
+        assert bytes("Protocol  : {}".format(openssl_version).encode('utf-8')) in results.stdout
+
+    # The server should indicate a session has been resumed
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
```

Comment:
Add `assert results.stderr is None` to be sure no error messages were returned

Reply:
https://github.com/aws/s2n-tls/pull/2814#discussion_r630447108

Reply:
Actually the `server` here is S2N and not OpenSSL. S2N does not print messages to caller in stderr like OpenSSL does. For S2N, `results.stderr != None` means we have an error.

Reply:
Added these asserts to s2n results